Source/JavaScriptCore/yarr/YarrJIT.cpp

```diff
@@ -1993,252 +1994,373 @@ class YarrGenerator final : public YarrJITInfo {
         // m_ops always ends with a YarrOpCode::BodyAlternativeEnd or YarrOpCode::MatchFailed
         // node, so there must always be at least one more node.
         ASSERT(opIndex + 1 < m_ops.size());
-        YarrOp* nextOp = &m_ops[opIndex + 1];
-
-        PatternTerm* term = op.m_term;
-        char32_t ch = term->patternCharacter;
-
-        if (!isLatin1(ch) && (m_charSize == CharSize::Char8)) {
-            // Have a 16 bit pattern character and an 8 bit string - short circuit
-            defaultMatchTargets.appendFailed(m_jit.jump());
-            return;
-        }
 
         const MacroAssembler::RegisterID character = m_regs.regT0;
 #if CPU(X86_64) || CPU(ARM64) || CPU(RISCV64)
         unsigned maxCharactersAtOnce = m_charSize == CharSize::Char8 ? 8 : 4;
 #else
         unsigned maxCharactersAtOnce = m_charSize == CharSize::Char8 ? 4 : 2;
 #endif
-        uint64_t ignoreCaseMask = 0;
-#if CPU(BIG_ENDIAN)
-        uint64_t allCharacters = ch << (m_charSize == CharSize::Char8 ? 24 : 16);
-#else
-        uint64_t allCharacters = ch;
-#endif
-        unsigned numberCharacters;
-        unsigned startTermPosition = term->inputPosition;
+
+        uint64_t charMask = m_charSize == CharSize::Char8 ? 0xff : 0xffff;
+        Vector<YarrOp*, 16> opList;
+        unsigned firstPosition = op.m_term->inputPosition;
+        unsigned lastPosition = firstPosition;
+        auto firstChar = op.m_term->patternCharacter;
+        bool have16BitCharacter = !isLatin1(firstChar);
 
         // For case-insesitive compares, non-ascii characters that have different
         // upper & lower case representations are converted to a character class.
-        ASSERT(!op.m_term->ignoreCase() || isASCIIAlpha(ch) || isCanonicallyUnique(ch, m_canonicalMode));
+        ASSERT(!op.m_term->ignoreCase() || isASCIIAlpha(firstChar) || isCanonicallyUnique(firstChar, m_canonicalMode));
 
-        if (op.m_term->ignoreCase() && isASCIIAlpha(ch)) {
+        if (have16BitCharacter && m_charSize == CharSize::Char16 && !U_IS_BMP(firstChar)) {
+            // The first term we are considering is a non-BMP char in a 16 bit pattern. Just try matching it and be done.
 #if CPU(BIG_ENDIAN)
```

Comment:
We no longer support BigEndian JIT architecture, so maybe we can just remove this code.

Reply:
I will gladly remove the BIG_ENDIAN code.